### PR TITLE
Split blockly in two packages

### PR DESCRIPTION
**Pull request by hbcarlos**
Splits the TypeScript package into two packages, the first one, jupyterlab-blockly includes the different classes to create the editor, etc., and the second one, jupyterlab-blockly-extension includes only the plugin for jupyterlab.

### CODE REVIEW

File: packages/extension/package.json [hunk 1]
```diff
@@ -0,0 +1,83 @@
+{
```
Comment (jtpio) [inline packages/extension/package.json]: Maybe the `packages/extension` folder could be renamed to `packages/blockly-extension` for consistency?

Comment (hbcarlos) [reply packages/extension/package.json]: Yep, that makes more sense. Thanks

**Comment by hbcarlos** [comment]
This is ready for review @jtpio. I didn't change anything for the release, I believe it should work as it is. Let me know if I should change something.

**Comment by jtpio** [comment]
It's possible the versions of the packages are not bumped properly.

Just checked the releaser assets and the packages still are on `0.1.0-alpha.2`:

image

We might need to use a custom `version-cmd` for the releaser, which would look like the following: `jlpm run lerna version --no-push --force-publish --no-git-tag-version`

File: packages/blockly-extension/package.json [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  ],
+  "styleModule": "style/index.js",
+  "publishConfig": {
+    "access": "public"
+  },
+  "jupyterlab": {
+    "extension": true,
+    "outputDir": "../../jupyterlab_blockly/labextension"
+  },
+  "jupyter-releaser": {
```
Comment (jtpio) [inline packages/blockly-extension/package.json]: This config should be moved to the top-level `package.json` or to `pyproject.toml`.

Comment (jtpio) [reply packages/blockly-extension/package.json]: And we can define a custom `version-cmd` to make sure all the packages are correctly bumped: `jlpm run lerna version --no-push --force-publish --no-git-tag-version`.

Normally the Python version is still computed from the JS version in the `setup.py`, so only the JS packages would need to be bumped.

Comment (hbcarlos) [reply packages/blockly-extension/package.json]: > This config should be moved to the top-level package.json or to pyproject.toml.

What do you mean? The release config, the extension metadata, or both?

Comment (hbcarlos) [reply packages/blockly-extension/package.json]: And, regarding the `version-cmd` do I need to add a release option to tell the release what command to execute, or will the release run the `version-cmd` command by default?

Comment (jtpio) [reply packages/blockly-extension/package.json]: > What do you mean? The release config, the extension metadata, or both?

Just the `jupyter-releaser`.



> And, regarding the `version-cmd` do I need to add a release option to tell the release what command to execute, or will the release run the `version-cmd` command by default?

Yes it's specified as a releaser option. For example: https://github.com/voila-dashboards/voila-gridstack/blob/130e4671493e123c9be3502337cdd85e17841670/pyproject.toml#L38-L39

Comment (hbcarlos) [reply packages/blockly-extension/package.json]: Ooh, got it. Thanks, @jtpio!

**Comment by jtpio** [comment]
Probably `setup.py` should read the version from `packages/blockly-extension/package.json` here instead of the top-level `package.json`?

https://github.com/QuantStack/jupyterlab-blockly/blob/5181e91e9fc5c8334bad894a09fe8a60dfb8216d/setup.py#L32-L33

**Comment by jtpio** [comment]
And the version in the top-level `package.json` could be a placeholder, since it's `private` and not meant to be published.

**Comment by hbcarlos** [comment]
Thanks, @jtpio! Do you have any other tips?

**Comment by jtpio** [comment]
This is looking good!

Just checked the built assets and they are all on `0.1.0` now.

File: package.json [hunk 3]
```diff
@@ -1,108 +1,43 @@
 {
   "name": "jupyterlab-blockly",
```
Comment (jtpio) [inline package.json]: Maybe this one can be renamed to `jupyterlab-blockly-root` to differentiate it from `packages/blockly`?

Comment (hbcarlos) [reply package.json]: Yep, that makes sense. Thanks!

**Comment by hbcarlos** [comment]
Thanks for looking into this PR @jtpio!

**Pull request merged**